src/AutoVizuA11y.tsx

```diff
@@ -96,141 +106,111 @@ const AutoVizuA11y = ({
 	autoDescriptions,
 	children,
 }: AutoVizuA11yProps) => {
-	let chart = React.Children.map(children, (child) => <div>{child}</div>);
-	const ref = React.useRef<HTMLDivElement>(null);
-
-	//could be converted to prop
-	let alertDiv: HTMLElement | null;
-
-	let storedLonger: string | null;
-	let storedSmaller: string | null;
-
-	let toolTutorial;
-
-	let apiKey: string;
-	let model: string | undefined;
-	let temperature: number | undefined;
-
 	const [series, setSeries] = useState<string[]>([]);
 	const [selectedSeries, setSelectedSeries] = useState<string>("");
 	const [insightsArray, setInsightsArray] = useState<number[]>([]);
 	const [arrayConverted, setArrayConverted] = useState<number[]>([]);
 	const [number, setNumber] = useState<number>(1);
 	const [descs, setDescs] = useState<string[]>([]);
+	const [descriptionContent, setDescriptionContent] = useState<string>("Generating description...");
+	const [elements, setElements] = useState<HTMLElement[]>([]);
+
+	const chartRef = useRef<HTMLDivElement>(null);
+	const shortcutGuideRef = useRef<HTMLDivElement>(null);
+
+	let componentId = useAutoId();
+
+	let alertDivRef = useRef<HTMLDivElement>(null);
+	let alertDiv: React.ReactNode = useMemo(
+		() => (
+			<div
+				ref={alertDivRef}
+				className="a11y_alert visually-hidden"
+				role="alert"
+				aria-live="assertive"
+			>
+				{"\u00A0"}
+			</div>
+		),
+		[],
+	);
 
-	if (autoDescriptions) {
-		apiKey = autoDescriptions.apiKey;
-		model = autoDescriptions.model;
-		temperature = autoDescriptions.temperature;
-	}
-
-	if (insights == undefined || !(insights in data[0])) {
-		insights = "";
-	}
-
-	// Generate a unique identifier for this component
-	const componentId = newId();
-
-	const storedLongerKey = `oldLonger_${componentId}`;
-	const storedSmallerKey = `oldSmaller_${componentId}`;
-
-	const handleFocus = () => {
-		//css
-		ref.current!.classList.add("focused");
-		toolTutorial = localStorage.getItem("toolTutorial");
-		if (toolTutorial === "true") {
-			alertDiv = ref.current!.getElementsByClassName("a11y_alert")[0] as HTMLElement | null;
-			if (alertDiv) {
-				alertDiv.textContent =
-					"You just entered an AutoVizually chart." +
-					" For information on how to interact with it, press the question mark key" +
-					" to open the shortcut guide";
+	let chartDescription: React.ReactNode = useMemo(
+		() => (
+			<p
+				style={{ textIndent: "-10000px" }}
+				className="a11y_desc visually-hidden"
+				data-testid="a11y_desc"
+				onFocus={() => {
```

Comment:
This can be moved to a callback using `useCallback` to avoid re-creating this function on every render.

```suggestion
				onFocus={handleFirstFocus}
```